command/pipe.go

```diff
@@ -0,0 +1,355 @@
+		partSize:         c.Int64("part-size") * megabytes,
+		encryptionMethod: c.String("sse"),
+		encryptionKeyID:  c.String("sse-kms-key-id"),
+		acl:              c.String("acl"),
+		cacheControl:     c.String("cache-control"),
+		expires:          c.String("expires"),
+		contentType:      c.String("content-type"),
+		contentEncoding:  c.String("content-encoding"),
+		dstRegion:        c.String("destination-region"),
+
+		// s3 options
+		storageOpts: NewStorageOpts(c),
+	}, nil
+}
+
+// Run starts copying stdin output to destination.
+func (c Pipe) Run(ctx context.Context) error {
+	var task = c.prepareUploadTask(ctx, c.dst)
+	var err = task()
+
+	return err
+}
+
+func (c Pipe) prepareUploadTask(
+	ctx context.Context,
+	dsturl *url.URL,
+) func() error {
+	return func() error {
+		if dsturl.IsBucket() || dsturl.IsPrefix() {
+			return fmt.Errorf("target %q must be a file", dsturl)
+		}
+
+		err := c.doUpload(ctx, dsturl)
+		if err != nil {
+			return &errorpkg.Error{
+				Op:  c.op,
+				Src: nil,
+				Dst: dsturl,
+				Err: err,
+			}
+		}
+		return nil
+	}
+}
+
+func (c Pipe) doUpload(ctx context.Context, dsturl *url.URL) error {
+	err := c.shouldOverride(ctx, dsturl)
+	if err != nil {
+		if errorpkg.IsWarning(err) {
+			printDebug(c.op, err, nil, dsturl)
+			return nil
+		}
+		return err
+	}
+
+	// override destination region if set
+	if c.dstRegion != "" {
+		c.storageOpts.SetRegion(c.dstRegion)
+	}
+	dstClient, err := storage.NewRemoteClient(ctx, dsturl, c.storageOpts)
+	if err != nil {
+		return err
+	}
+
+	metadata := storage.NewMetadata().
+		SetStorageClass(string(c.storageClass)).
+		SetSSE(c.encryptionMethod).
+		SetSSEKeyID(c.encryptionKeyID).
+		SetACL(c.acl).
+		SetCacheControl(c.cacheControl).
+		SetExpires(c.expires)
+
+	if c.contentType != "" {
+		metadata.SetContentType(c.contentType)
+	} else {
+		metadata.SetContentType(guessContentTypeByExtension(dsturl))
+	}
+
+	if c.contentEncoding != "" {
+		metadata.SetContentEncoding(c.contentEncoding)
+	}
+
+	err = dstClient.Put(ctx, &stdin{file: os.Stdin}, dsturl, metadata, c.concurrency, c.partSize)
+
+	if err != nil {
+		return err
+	}
+
+	msg := log.InfoMessage{
+		Operation:   c.op,
+		Source:      nil,
+		Destination: dsturl,
+		Object: &storage.Object{
+			StorageClass: c.storageClass,
+		},
+	}
+	log.Info(msg)
+
+	return nil
+}
+
+// shouldOverride function checks if the destination should be overridden if
+// the destination object and given pipe flags conform to the
+// override criteria.
+func (c Pipe) shouldOverride(ctx context.Context, dsturl *url.URL) error {
+	// if not asked to override, ignore.
+	if !c.noClobber {
+		return nil
+	}
+
+	dstClient, err := storage.NewClient(ctx, dsturl, c.storageOpts)
+	if err != nil {
+		return err
+	}
+
+	dstObj, err := getObject(ctx, dsturl, dstClient)
+	if err != nil {
+		return err
+	}
+
+	// if destination not exists, no conditions apply.
+	if dstObj == nil {
+		return nil
+	}
+
+	var stickyErr error
```

Comment:
`stickyErr` is not sticky. Please remove the variable and return the error directly in 2 lines below.